yb-voyager/src/srcdb/data/gather-assessment-metadata/postgresql/table-row-counts.psql

```diff
@@ -1,42 +1,17 @@
--- Create a temporary table to store the row counts
-CREATE TEMP TABLE temp_table (
-    schema_name TEXT,
-    table_name TEXT,
-    row_count BIGINT
-);
-
 -- Set the schema_list variable
 SET vars.schema_list TO :'schema_list';
 
-DO $$
-DECLARE
-    schema_list TEXT[] := string_to_array(current_setting('vars.schema_list'), '|');
-    schema_name TEXT;
-    table_record RECORD;
-BEGIN
-    -- Iterate over each schema in the schema list
-    FOREACH schema_name IN ARRAY schema_list LOOP
-        -- Iterate over each table in the current schema
-        FOR table_record IN 
-            SELECT s.relname AS table_name,
-                   n.nspname AS table_schema,
-                   COALESCE(NULLIF(s.n_live_tup, -1), 0) AS estimated_row_count
-            FROM pg_stat_user_tables s
-            JOIN pg_namespace n ON s.schemaname = n.nspname
-            WHERE n.nspname = schema_name
-        LOOP
-            -- Insert estimated row count data into temp_table
-            EXECUTE format(
-                'INSERT INTO temp_table (schema_name, table_name, row_count) VALUES (%L, %L, %s)',
-                table_record.table_schema,
-                table_record.table_name,
-                table_record.estimated_row_count
-            );
-        END LOOP;
-    END LOOP;
-END $$;
-
--- Export the data from the temporary table to a CSV file
-\copy temp_table TO 'table-row-counts.csv' WITH CSV HEADER;
-
-DROP TABLE temp_table;
\ No newline at end of file
+-- Iterate over each schema in the schema list
+-- (Original DO block inserted row counts; this SELECT returns the same rows without writes)
+\o table-row-counts.csv
+COPY (
+    SELECT
+        :'source_node_name' AS source_node,
+        n.nspname AS schema_name,
+        s.relname AS table_name,
+        COALESCE(NULLIF(s.n_live_tup, -1), 0) AS row_count
+    FROM pg_stat_user_tables s
+    JOIN pg_namespace n ON s.schemaname = n.nspname
+    WHERE n.nspname = ANY(ARRAY[string_to_array(:'schema_list', '|')])
+) TO STDOUT WITH CSV HEADER;
+\o
```

Comment:
is this just to simplify the previous appraoch ?

Reply:
Yep, the script originally used a temp table plus a DO loop to iterate schemas and insert row-counts, but this failed on read replicas because temp tables are not allowed and DO blocks cannot substitute psql variables. 

The logic inside the loop was unnecessary anyway, since the same data comes directly from pg_stat_user_tables, so the entire DO block and inserts were redundant. 

The current version works because it uses a single static SELECT with WHERE schema IN (…), which produces identical results.